src/test/munit/response-streaming/selectors-not-being-blocked-when-response-streaming.xml

```diff
@@ -0,0 +1,81 @@
+        http://www.mulesoft.org/schema/mule/java http://www.mulesoft.org/schema/mule/java/current/mule-java.xsd">
+
+    <munit:config name="selectors-not-being-blocked-when-response-streaming" minMuleVersion="4.3.0"/>
+
+    <munit:dynamic-port propertyName="dynamic.port"/>
+
+    <http:listener-config name="Listener_config" doc:name="HTTP Listener config" >
+        <http:listener-connection host="0.0.0.0" port="${dynamic.port}" />
+    </http:listener-config>
+
+    <http:request-config name="HTTP_Request_config" >
+        <http:request-connection host="localhost" port="${dynamic.port}"/>
+    </http:request-config>
+
+    <munit:test name="Some selector is available while streaming" description="Test">
+        <munit:enable-flow-sources >
+            <munit:enable-flow-source value="TheStreamingFlow" />
+            <munit:enable-flow-source value="OtherFlow" />
+        </munit:enable-flow-sources>
+        <munit:execution >
+            <set-variable doc:name="Set number of requests" variableName="requestsNumber" value="#[%dw 2.0
+import java!org::mule::test::PipedStreamGenerator
+output application/java
+---
+PipedStreamGenerator::availableProcessors()]"/>
+            <logger level="INFO" doc:name="Logger" message='#["Will send " ++ vars.requestsNumber ++ " requests"]'/>
+            <flow-ref doc:name="Spawn requests" name="SpawnRequests"/>
+            <java:invoke-static doc:name="Wait requests started" class="org.mule.test.PipedStreamGenerator" method="waitForSources()" />
+            <java:invoke-static doc:name="Send a chunk" class="org.mule.test.PipedStreamGenerator" method="writeChunkInStreams()"/>
+            <java:invoke-static method="writeChunkInStreams()" doc:name="Send another chunk" class="org.mule.test.PipedStreamGenerator" />
+            <http:request method="GET" doc:name="Dummy Request" path="/dummy" config-ref="HTTP_Request_config"/>
+            <set-variable value="#[payload as String]" doc:name="Set Variable" variableName="dummy_response"/>
+            <java:invoke-static doc:name="Close streams" class="org.mule.test.PipedStreamGenerator" method="closeStreams()"/>
+        </munit:execution>
+        <munit:validation>
+            <munit-tools:assert-equals doc:name="Assert dummy request completed (some selector was available)" actual="#[vars.dummy_response]" expected='#["Dummy payload"]'/>
+        </munit:validation>
+    </munit:test>
+    <sub-flow name="SpawnRequests" >
+        <choice doc:name="Choice" >
+            <when expression="#[vars.requestsNumber &gt; 0]">
+                <async doc:name="Request async" >
+                    <http:request method="GET" doc:name="Request" path="/stream" config-ref="HTTP_Request_config" requestStreamingMode="ALWAYS" />
+                </async>
+                <set-variable value="#[vars.requestsNumber-1]" doc:name="Decrease number of requests" variableName="requestsNumber" />
+                <flow-ref doc:name="Spawn N-1 requests" name="SpawnRequests" />
```

Comment:
why not a `for-each` instead of using recursion?

Reply:
If it walks like a duck and it quacks like a duck, then it must be a duck